src/main/java/step2/StringCalculator.java

```diff
@@ -0,0 +1,59 @@
+package step2;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+
+    public static final int ZERO = 0;
+
+    public static int splitAndSum(String text) {
+        if (isNullAndBlank(text)) {
+            return ZERO;
+        }
+        return sum(toInts(strSplit(text)));
+    }
+
+    private static String[] strSplit(String text) {
+        Matcher m = getMatcher(text);
+        if (m.find()) {
+            String customDelimiter = m.group(1);
+            return m.group(2).split(customDelimiter);
+        }
+        return text.split("[,:]");
+    }
+
+    private static Matcher getMatcher(String text) {
+        return Pattern.compile("//(.)\n(.*)").matcher(text);
+    }
+
+    private static boolean isNullAndBlank(String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private static int sum(int[] intValues) {
+        int sum = ZERO;
+
+        for (int intValue : intValues) {
+            sum += intValue;
+        }
+        return sum;
+    }
+
+    private static int[] toInts(String[] strValues) {
+        int[] intValues = new int[strValues.length];
+        for (int i = 0; i < strValues.length; i++) {
+            int num = toPositiveNumber(strValues[i]);
+            intValues[i] = num;
+        }
+        return intValues;
+    }
```

Comment:
배열을 다루기 위해 크기도 지정하고 인덱스에 하나하나씩 넣어주는 작업들이 귀찮지 않으신가요?
Java에서 제공하는 컬렉션 중 List를 적극적으로 사용해 보시는 것을 추천드려요!

```suggestion
    private static List<Integer> toInts(String[] strValues) {
        List<Integer> numbers = new ArrayList<Integer>;
        for (String value : strValues) {
            numbers.add(toPositiveNumber(value));
        }
        return numbers;
    }
```